pkg/cloudprovider/providers/gce/gce_address_manager_test.go

```diff
@@ -0,0 +1,188 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gce
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	computebeta "google.golang.org/api/compute/v0.beta"
+)
+
+const svcNm = "my-service"
+const region = "us-central1"
+const subnet = "/projects/x/regions/us-central1/subnetworks/customsub"
+
+// TestAddressManagerNoRequestedIP tests the typical case of passing in no requested IP
+func TestAddressManagerNoRequestedIP(t *testing.T) {
+	svc := NewFakeCloudAddressService()
+	loadBalancerName := "a111111111111111"
+	targetIP := ""
+
+	mgr := newAddressManager(svc, svcNm, region, subnet, loadBalancerName, targetIP, schemeInternal)
+	ipToUse, err := mgr.HoldAddress()
+	assert.NoError(t, err)
+	assert.NotEmpty(t, ipToUse)
+
+	addr, err := svc.GetRegionAddress(loadBalancerName, region)
+	assert.NoError(t, err)
```

Comment:
Use `require.NoError(t, err)` to make sure the test fail, otherwise you may be deferencing a nil pointer `addr.Address`